uefi/flash.go

```diff
@@ -35,6 +37,57 @@ type FlashDescriptor struct {
 	ExtractPath string
 }
 
+func findSignature(buf []byte) (int, error) {
+	if bytes.Equal(buf[16:16+len(FlashSignature)], FlashSignature) {
+		// 16 + 4 since the descriptor starts after the signature
+		return 20, nil
+	}
+	if bytes.Equal(buf[:len(FlashSignature)], FlashSignature) {
+		// + 4 since the descriptor starts after the signature
+		return 4, nil
+	}
+	return -1, fmt.Errorf("Flash signature not found: first 20 bytes are:\n%s",
+		hex.Dump(buf[:20]))
+}
+
+// ParseFlashDescriptor parses the ifd from the buffer
+func (fd *FlashDescriptor) ParseFlashDescriptor() error {
+	if buflen := len(fd.buf); buflen != FlashDescriptorLength {
+		return fmt.Errorf("flash descriptor length not %#x, was %#x", FlashDescriptorLength, buflen)
+	}
+
+	descriptorMapStart, err := findSignature(fd.buf)
+	if err != nil {
+		return err
+	}
+	fd.DescriptorMapStart = uint(descriptorMapStart)
+
+	// Descriptor Map
+	desc, err := NewFlashDescriptorMap(fd.buf[fd.DescriptorMapStart : fd.DescriptorMapStart+FlashDescriptorMapSize])
+	if err != nil {
+		return err
+	}
+	fd.DescriptorMap = *desc
```

Comment:
Why isn't DescriptorMap just a pointer too? just wondering. This deref/copy seems more work than needed.

Reply:
ok

Reply:
done